cpp/src/arrow/filesystem/filesystem.h

```diff
@@ -132,8 +132,14 @@ struct ARROW_EXPORT FileSelector {
   bool recursive;
   /// The maximum number of subdirectories to recurse into.
   int32_t max_recursion;
-
-  FileSelector() : allow_not_found(false), recursive(false), max_recursion(INT32_MAX) {}
+  /// If true, use stat() to obtain extended file metadata.
```

Comment:
Since the default is `true`, maybe better to frame as what happens if false. Also maybe preferable to frame in implementation-agnostic way (stat isn't invoked on Windows, and isn't applicable to remote filesystems):
```suggestion
  /// If false, may skip retrieving size and modification time metadata.
  ///
  /// On local filesystem, setting to false avoids an additional system call.
  /// On other filesystems, this may have no effect.
```